encoding/rlp/rlp.go

```diff
@@ -0,0 +1,342 @@
+package rlp
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/bits"
+	"reflect"
+	"strconv"
+)
+
+const (
+	EmptyString = 0x80
+	EmptySlice  = 0xC0
+)
+
+type Code uint32
+
+const (
+	ErrUnknown Code = iota
+	ErrNoInput
+	ErrInvalid
+)
+
+func (c Code) Error() string {
+	switch c {
+	case ErrNoInput:
+		return "no input"
+	case ErrInvalid:
+		return "input not valid RLP encoding"
+	default:
+		return "unknown error"
+	}
+}
+
+func encodeUint8(input uint8) ([]byte, error) {
+	if input == 0 {
+		return []byte{EmptyString}, nil
+	} else if input >= 0x00 && input <= 0x7f {
+		return []byte{input}, nil
+	} else if input >= 0x80 && input <= 0xff {
+		return []byte{0x81, input}, nil
+	}
+	return []byte{EmptyString}, nil
+}
+
+func encodeUint64(i uint64) ([]byte, error) {
+	size := bits.Len64(i)/8 + 1
+	if size == 1 {
+		return encodeUint8(uint8(i))
+	}
+	b := make([]byte, 8)
+	binary.BigEndian.PutUint64(b, uint64(i))
+	return encodeString(b[8-size:])
+}
+
+func encodeLength(n, offset int) []byte {
+	if n <= 55 {
+		return []uint8{uint8(n + offset)}
+	}
+
+	length := strconv.FormatUint(uint64(n), 16)
+	if len(length)%2 == 1 {
+		length = fmt.Sprintf("0%s", length)
+	}
+
+	dec, err := hex.DecodeString(length)
+	if err != nil {
+		return nil
+	}
```

Comment:
I'm confused.  strconv.FormatUint(() produces a hex string, which you prepend with 0 if the length is odd. Then you hex decode the result. Is the answer going to be n?